crates/parcel_plugin_transformer_js/src/transformer.rs

```diff
@@ -1,24 +1,894 @@
+        invalidate_on_file_change.push(file_path);
+      }
+    }
+  }
+  Ok((dependency_by_specifier, invalidate_on_file_change))
+}
+
+fn make_export_star_symbol(asset_id: u64) -> Symbol {
+  Symbol {
+    exported: "*".into(),
+    // This is the mangled exports name
+    local: format!("${:016x}$exports", asset_id).into(),
+    loc: None,
+    flags: SymbolFlags::empty(),
+  }
+}
+
+/// Convert `CollectImportedSymbol`, `ImportedSymbol` and `ExportedSymbol` into `Symbol`
+macro_rules! convert_symbol {
+  ($asset_file_path: ident, $symbol: ident, $flags: expr) => {
+    Symbol {
+      exported: $symbol.imported.as_ref().into(),
+      local: $symbol.local.as_ref().into(),
+      loc: Some(convert_loc($asset_file_path.to_owned(), &$symbol.loc)),
+      flags: $flags,
+    }
+  };
+}
+
+/// Convert from `[CollectImportedSymbol]` to `[Symbol]`
+fn transformer_collect_imported_symbol_to_symbol(
+  asset_file_path: &Path,
+  symbol: &parcel_js_swc_core::CollectImportedSymbol,
+) -> Symbol {
+  convert_symbol!(asset_file_path, symbol, SymbolFlags::empty())
+}
+
+/// Convert from `[ImportedSymbol]` to `[Symbol]`
+fn transformer_imported_symbol_to_symbol(
+  asset_file_path: &Path,
+  symbol: &parcel_js_swc_core::ImportedSymbol,
+) -> Symbol {
+  convert_symbol!(asset_file_path, symbol, SymbolFlags::empty())
+}
+
+/// Convert from `[ExportedSymbol]` to `[Symbol]`
+fn transformer_exported_symbol_into_symbol(
+  asset_file_path: &PathBuf,
+  symbol: &parcel_js_swc_core::ExportedSymbol,
+) -> Symbol {
+  let mut flags = SymbolFlags::empty();
+  flags.set(SymbolFlags::IS_ESM, symbol.is_esm);
+  convert_symbol!(asset_file_path, symbol, flags)
+}
+
+fn make_esm_helpers_dependency(
+  options: &ParcelOptions,
+  asset_file_path: &PathBuf,
+  asset_environment: Environment,
+  dependency_flags: DependencyFlags,
+  asset_id: u64,
+) -> Dependency {
+  Dependency {
+    source_asset_id: Some(format!("{:016x}", asset_id)),
+    specifier: "@parcel/transformer-js/src/esmodule-helpers.js".into(),
+    specifier_type: SpecifierType::Esm,
+    source_path: Some(asset_file_path.clone()),
+    env: Environment {
+      include_node_modules: IncludeNodeModules::Map(
+        [("@parcel/transformer-js".to_string(), true)]
+          .into_iter()
+          .collect(),
+      ),
+      ..asset_environment.clone()
+    }
+    .into(),
+    resolve_from: Some(options.core_path.as_path().into()),
+    range: None,
+    priority: Priority::Sync,
+    bundle_behavior: BundleBehavior::None,
+    flags: dependency_flags,
+    loc: None,
+    // placeholder: None,
+    target: None,
+    // promise_symbol: None,
+    symbols: Vec::new(),
+    // import_attributes: Vec::new(),
+    pipeline: None,
+    meta: JSONObject::new(),
+    // resolver_meta: JSONObject::new(),
+    package_conditions: ExportsCondition::empty(),
+    // custom_package_conditions: Vec::new(),
+    // TODO:
+    is_entry: false,
+    needs_stable_name: false,
+    is_optional: false,
```

Comment:
I reckon these fields are deprecated by `AssetFlags` and such

Reply:
Yeah they are

Reply:
Yeah they are